### PR TITLE
validate noobaa status

### PR DESCRIPTION
**Pull request by bernerhat**
validate the the non optional components of noobaa. (issue #1871 )

### CODE REVIEW

File: ocs_ci/ocs/cluster.py [hunk 1]
```diff
@@ -189,7 +189,7 @@ def is_health_ok(self):
         self.cluster.reload()
         return self.cluster.data['status']['ceph']['health'] == "HEALTH_OK"
 
-    def cluster_health_check(self, timeout=None):
+    def cluster_health_check(self, mcg_obj_session, timeout=None):
```
Comment (ebenahar) [inline ocs_ci/ocs/cluster.py:192]: is `mcg_obj_session` a fixture?
if so, this usage is wrong as cluster_health_check is not a test function. Pytest recognized this usage only with a function used for the test (i.e, `test_`)

Comment (bernerhat) [reply ocs_ci/ocs/cluster.py]: changed to initialize an MCG object within the class

File: ocs_ci/ocs/resources/mcg.py [hunk 2]
```diff
@@ -710,3 +710,20 @@ def exec_mcg_cmd(self, cmd, namespace=None, **kwargs):
...
         result.stderr = result.stderr.decode()
         return result
+
+    @property
+    def status(self):
+        """
+        Verify noobaa status output is clean without any errors
+
+        Returns:
+            bool: return False if any of the non optional components of noobaa is not available
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/mcg.py:720]: add an empty line below

File: ocs_ci/ocs/resources/mcg.py [hunk 3]
```diff
@@ -48,17 +48,17 @@ def __init__(self):
...
-            .get('serviceS3').get('externalDNS')[-1]
+                .get('serviceS3').get('externalDNS')[-1]
         )
         self.mgmt_endpoint = (
-            results.get('items')[0].get('status').get('services')
-            .get('serviceMgmt').get('externalDNS')[-1]
-        ) + '/rpc'
+                                 results.get('items')[0].get('status').get('services')
+                                     .get('serviceMgmt').get('externalDNS')[-1]
+                             ) + '/rpc'
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/mcg.py:56]:
```suggestion
self.mgmt_endpoint = (
    results.get('items')[0].get('status').get('services')
    .get('serviceMgmt').get('externalDNS')[-1]
) + /rpc'
```

File: ocs_ci/ocs/resources/mcg.py [hunk 4]
```diff
@@ -48,17 +48,17 @@ def __init__(self):
...
         self.region = config.ENV_DATA['region']
 
         creds_secret_name = (
             results.get('items')[0].get('status').get('accounts')
-            .get('admin').get('secretRef').get('name')
+                .get('admin').get('secretRef').get('name')
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/mcg.py:61]: delete extra indentation

File: ocs_ci/ocs/resources/mcg.py [hunk 5]
```diff
@@ -121,7 +122,8 @@ def __init__(self):
                 aws_secret_access_key=self.aws_access_key
             )
             logger.info('Checking whether RGW pod is not present on AWS platform')
-            pods = pod.get_pods_having_label(label=constants.RGW_APP_LABEL, namespace=self.namespace)
+            pods = pod.get_pods_having_label(label=constants.RGW_APP_LABEL,
+                                             namespace=self.namespace)
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/mcg.py:126]:
```suggestion
            pods = pod.get_pods_having_label(
                label=constants.RGW_APP_LABEL, namespace=self.namespace
            )
```

**Comment by bernerhat** [comment]
> PR validation [Job PASSED](https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-trigger-test-pr/1782//testReport).

aws deployment

File: ocs_ci/ocs/resources/mcg.py [hunk 6]
```diff
@@ -710,3 +710,22 @@ def exec_mcg_cmd(self, cmd, namespace=None, **kwargs):
...
+        Verify noobaa status output is clean without any errors
+
+        Returns:
+            bool: return False if any of the non optional components of noobaa is not available
+
+        """
+        # Get noobaa status
+        status = self.exec_mcg_cmd('status').stderr
+        for line in status.split('\n'):
+            if 'Not Found' in line:
```
Comment (MeridianExplorer) [inline ocs_ci/ocs/resources/mcg.py:726]: I think we can switch to short circuiting for better readability - 
`if 'Not Found' in line and 'Optional' not in line:`
Not a must, though

**Comment by bernerhat** [comment]
> PR validation [Job PASSED](https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-trigger-test-pr/1833//testReport).

vmware deployment

**Comment by ebenahar** [comment]
Let's wait with merging until we finish OCS 4.4 RC7 testing

**Pull request merged**